rust/core-lib/src/file.rs

```diff
@@ -134,13 +139,16 @@ impl FileManager {
     }
 
     fn save_new(&mut self, path: &Path, text: &Rope, id: BufferId) -> Result<(), FileError> {
-        try_save(path, text, CharacterEncoding::Utf8)
+        try_save(path, text, CharacterEncoding::Utf8, self.get_info(id).unwrap())
```

Comment:
The `unwrap` here causes a panic when creating a new file in xi editor and saving it.

Reply:
Hm, should I guess I'll just optionalise FileInfo then, there's nothing to keep when we create a new file anyway.